sdk/core/azure-json/src/main/java/com/azure/json/JsonArray.java

```diff
@@ -0,0 +1,373 @@
+//        // appending each JsonElement to jsonOutput
+//        for(Iterator<JsonElement> itr = this.elements.iterator(); itr.hasNext();) {
+//            // Get the next JsonElement from the elements JsonArray object
+//            JsonElement element = itr.next();
+//
+//            // Case: element is a JsonObject, therefore the whole JSON object can
+//            // be stringified via its toJson method, then appended to jsonOutput
+//            if(element instanceof JsonObject) {
+//                jsonOutput += ((JsonObject)element).toJson();
+//            }
+//            // Case: element is a JsonArray, therefore the nested JSON array can
+//            // be stringified via recursively calling toJson and then appending
+//            // the returned String to jsonOutput
+//            else if(element instanceof JsonArray) {
+//                jsonOutput += ((JsonArray)element).toJson();
+//            }
+//            // Case: element is not a JsonObject or JsonArray, therefore it can
+//            // be simply converted to a String object and appended to jsonOutput
+//            else { jsonOutput += element; }
+//
+//            // Case: haven't reached the end of the JSON array, therefore more
+//            // elements must be converted to String objects and appended to the
+//            // jsonOutput
+//            // Elements are separated by commas.
+//            if(itr.hasNext()) { jsonOutput += ", "; }
+//        }
+//
+//        // Returning the resulting String representation of the JsonArray object
+//        // JSON arrays are delimited by opening and closing square brackets.
+//        return "[" + jsonOutput + "]";
+//    }
+
+    /**
+     * @return boolean of whether this JsonElement object is of type JsonArray.
+     */
+    @Override
+    public boolean isArray() { 
+        return true; 
+    }
+
+    @Override
+    public JsonArray asArray() { 
+        return this;
+    }
+
+    @Override
+    public JsonObject asObject() {
+        JsonObject output = new JsonObject();
+        for (int i = 0; i < elements.size(); i++) {
+            String keyword = "Value";
+            if (i > 0) {
+                keyword += i;
+            }
+            output.addProperty(keyword, elements.get(i));
+        }
+        return output;
+    }
+
+    @Override
+    public JsonBoolean asBoolean() {
+        if (elements.size() >= 1) {
+            return elements.get(0).asBoolean();
+        } else {
+            return JsonBoolean.getInstance(true);
+        }
+    }
+
+    @Override
+    public JsonNumber asNumber() {
+        if (elements.size() >= 1) {
+            return elements.get(0).asNumber();
+        } else {
+            return new JsonNumber();
+        }
+    }
+
+    @Override
+    public JsonString asString() {
+        if (elements.size() >= 1) {
+            return elements.get(0).asString();
+        } else {
+            return new JsonString();
+        }
+    }
```

Comment:
@JM16450251 as discussed in chat with @alzimmermsft @JonathanGiles - basically contextualise the 'as' methods, so that they are related (i.e JsonObject -> map, JsonArray -> list). All should have a to string.

public abstract class JsonElement<T> {
    public T getValue();
}
public class JsonNumber extends JsonElement<Number> {
    public Number getValue();
}